code/datums/status_effects/buffs.dm

```diff
@@ -559,3 +559,21 @@
 	. = ..()
 	to_chat(owner, span_warning("You don't feel any healthier."))
 
+/datum/status_effect/reagent_massage
+	id = "reagent_massage"
+	duration = 10 MINUTES
+	status_type = STATUS_EFFECT_REPLACE
+	alert_type = /atom/movable/screen/alert/status_effect/reagent_massage
+	var/list/reagents_to_apply = list()
+	var/potency = 100
+
+/datum/status_effect/reagent_massage/tick()
+	if(ishuman(owner))
+		var/mob/living/carbon/human/human_owner = owner
+		for(var/reagent in reagents_to_apply)
+			human_owner.reagents.add_reagent(reagent, 1*(potency/100))
```

Comment:
I would recommend capping the amount of volume this can give you to `1*(potency/100)` or so. As it is at maximum rating and potency this will rapidly stack to hundreds of units.